### PR TITLE
docs: add Modern.js v2 release blog

### PR DESCRIPTION
**Pull request by chenjiahan**
## Description

- Add Blogs to document.
- Add Modern.js v2 release blog.

## Types of changes

- [x] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+大家好，很高兴地向大家宣布，Modern.js v2 版本已经正式发布了！
+
+**Modern.js 是字节跳动 Web Infra 团队开源的一套 Web 工程体系**。在开源以来的一年多时间里，Modern.js 保持稳定的迭代节奏，数十位贡献者参与了开发，累计提交 2000+ 个 Pull Request，并支持了 Rspack 构建、嵌套路由、流式渲染等新特性。
+
+在这篇文章里，我们会和大家一起聊一聊 Modern.js 在过去一年多时间里的变化。
+
+![Modern.js v2 官网](https://lf3-static.bytednsdoc.com/obj/eden-cn/zq-uylkvT/ljhwZthlaukjlkulzlp/v2-release/Framework-Home.png)
+
+<div style={{ textAlign: 'center', fontStyle: 'italic' }}>
```
Comment (zllkjc) [inline packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx:18]: 这里是不是缺了些 markdown 语法，我看下面也有类似的地方

File: packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+在 Modern.js v2 中，我们约定在 src/routes 目录下的文件，在应用启动时会自动生成对应的路由结构，页面的路由与文件结构是相呼应的，并且可以为资源加载带来一些性能优化。
+
+Modern.js 的嵌套路由包含以下功能：
+
+- **资源加载**：基于路由结构，并行加载多级路由资源，子组件资源加载不再受父组件渲染影响。并且将路由资源进一步细分，保证路由跳转时的最小资源加载。
+- **共享布局**：在路由之间轻松共享 UI，同时保持状态，并避免多余的重新渲染。
+- **数据获取**：基于路由结构完成多级路由的数据并行获取。
+
+在 Modern.js v2 中，只需要配置 runtime.router，即可开启嵌套路由：
```
Comment (zllkjc) [inline packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx:152]: runtime.router ---> `runtime.router`

File: packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+> 如果你对 Modern.js 的插件系统感兴趣，请阅读 [「Modern.js - 自定义插件」](https://modernjs.dev/guides/topic-detail/framework-plugin/introduction.html)文档。
+
+### 嵌套路由
+
+**Modern.js v2 基于 React Router v6 提供了新的路由方式 —— 嵌套路由（Nested Routes）。**
+
+如果大家了解过 Remix 或 Next.js 13，应该对嵌套路由不陌生了。嵌套路由与 Modern.js v1 提供的约定式路由相似，也是一种基于文件系统的路由。
+
+在 Modern.js v2 中，我们约定在 src/routes 目录下的文件，在应用启动时会自动生成对应的路由结构，页面的路由与文件结构是相呼应的，并且可以为资源加载带来一些性能优化。
```
Comment (zllkjc) [inline packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx:144]: src/routes ---> `src/routes`

File: packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx (hunk 2 above)
Comment (chenjiahan) [reply packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx]: done~

File: packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx (hunk 3 above)
Comment (chenjiahan) [reply packages/document/main-doc/docs/zh/blog/updates/v2-release-note.mdx]: done~

**Review by zllkjc — APPROVED** [review]

**Pull request merged**